src/Runtime/Distributions/Automata/Automaton.cs

```diff
@@ -277,13 +253,13 @@ public State Start
         [Construction("GetStates", "Start")]
         public static TThis FromStates(IEnumerable<State> states, State startState)
```

Comment:
Might again make sense to operate on `IReadOnlyCollection<State>` here and `SetStates` could have a `private` overload with the known capacity.

Reply:
This code is carried from

I have an overhaul of way states are stored in mind. Will do it then, because otherwise I will have to re-do it to adjust to new scheme.